test/cpp/ext/filters/census/stats_plugin_end2end_test.cc

```diff
@@ -730,6 +731,55 @@ TEST_F(StatsPluginEnd2EndTest, TestAllSpansAreExported) {
             recv_span_data->parent_span_id());
 }
 
+// Test the working of GRPC_ARG_DISABLE_OBSERVABILITY.
+TEST_F(StatsPluginEnd2EndTest, TestObservabilityDisabledChannelArg) {
+  {
+    // Client spans are ended when the ClientContext's destructor is invoked.
+    ChannelArguments args;
+    args.SetInt(GRPC_ARG_ENABLE_OBSERVABILITY, 0);
+    auto channel = CreateCustomChannel(server_address_,
+                                       InsecureChannelCredentials(), args);
+    ResetStub(channel);
+    EchoRequest request;
+    request.set_message("foo");
+    EchoResponse response;
+
+    grpc::ClientContext context;
+    ::opencensus::trace::AlwaysSampler always_sampler;
+    ::opencensus::trace::StartSpanOptions options;
+    options.sampler = &always_sampler;
+    auto sampling_span =
+        ::opencensus::trace::Span::StartSpan("sampling", nullptr, options);
+    grpc::CensusContext app_census_context("root", &sampling_span,
+                                           ::opencensus::tags::TagMap{});
+    context.set_census_context(
+        reinterpret_cast<census_context*>(&app_census_context));
+    traces_recorder_->StartRecording();
+    grpc::Status status = stub_->Echo(&context, request, &response);
+    EXPECT_TRUE(status.ok());
+  }
+  absl::SleepFor(absl::Milliseconds(500 * grpc_test_slowdown_factor()));
```

Comment:
Is there any way to do the same without sleep? This could be another source of making this test flaky.

Reply:
We've been using this 500ms sleep in this test for quite some time and I don't think this test has come up as flaky. 

That being said, I do want to remove the sleep from these tests. One way to do this would be to simply loop with a deadline till we get what we want, but if I'm doing this I'd like to do it for all the tests, and I would prefer not doing it in this PR

Reply:
Assuming that this will be covered by later PR, I approve this. Thanks!